echopop/spatial/transect.py

```diff
@@ -346,3 +363,130 @@ def transect_bearing(transect_data: pd.DataFrame):
 
     # Return the output
     return transect_direction.reset_index()
+
+
+def export_transect_layers(
+    transect_data: pd.DataFrame,
+    index_variable: Union[str, List[str]] = ["transect_num", "interval"],
+):
+    """
+    Calculate the mean depth and layer height for specific grouped data.
+    """
+
+    # Check that index variables exist
+    # ---- Convert to list, if needed
+    if isinstance(index_variable, str):
+        index_variable = list(index_variable)
+    elif not isinstance(index_variable, list):
+        raise TypeError(
+            f"The defined `region_filter` ({index_variable}) must be either a `str` or `list`."
+        )
+
+    # Check columns
+    # ---- Missing columns
+    missing_columns = set(index_variable) - set(transect_data.columns)
+    # ---- Raise error if needed
+    if missing_columns:
+        raise ValueError(
+            f"The following columns are missing from `transect_data`: {list(missing_columns)}"
+        )
+
+    # Calculate the maximum depth
+    if "max_depth" not in transect_data.columns:
+        raise ValueError("The column 'max_depth' is missing from `transect_data`.")
+    else:
+        # ---- Compute the bottom depth
+        transect_summary = (
+            transect_data.groupby(index_variable)["max_depth"].max().to_frame("bottom_depth")
+        )
+
+    # Calculate the mean layer depth and height
+    if "layer_depth_min" not in transect_data.columns:
+        raise ValueError("The column 'layer_depth_min' is missing from `transect_data`.")
+    elif "layer_depth_max" not in transect_data.columns:
+        raise ValueError("The column 'layer_depth_max' is missing from `transect_data`.")
+    else:
```

Comment:
This is fine. One way to do this is since the checks are all for columns, you could check the three first , and then if nothing fails all the computation follows. Something like:
```python
for col in ["max_depth", "layer_depth_min", "layer_depth_max"]:
   if col not in transect_data.columns:
      raise ValueError(f"...")

# If no errors are raised, proceed with computation
# ---- Compute the bottom depth
transect_summary = ...
# ---- Mean layer depth
...
```

This way the code is pretty clean without the many if-else.